src/libAtomVM/opcodesswitch.h

```diff
@@ -2213,7 +2213,16 @@ HOT_FUNC int scheduler_entry_point(GlobalContext *glb)
                 #endif
 
                 #ifdef IMPL_EXECUTE_LOOP
-                    int local_process_id = term_to_local_process_id(x_regs[0]);
+                    term process_term = x_regs[0];
+                    int local_process_id;
+                    if (term_is_atom(process_term)) {
```

Comment:
I think we should check if `recipient` is a `pid` first, after that if it is an atom and registered, and as last thing just raise badarg.